ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/restore/KsqlRestoreCommandTopic.java

```diff
@@ -394,30 +395,42 @@ private static void maybeCleanUpQuery(final byte[] command, final KsqlConfig ksq
     boolean queryIdFound = false;
     final Map<String, Object> streamsProperties =
         new HashMap<>(ksqlConfig.getKsqlStreamConfigProps());
+    boolean sharedRuntimeQuery = false;
+    String queryId = "";
     final JSONObject jsonObject = new JSONObject(new String(command, StandardCharsets.UTF_8));
     if (hasKey(jsonObject, "plan")) {
       final JSONObject plan = jsonObject.getJSONObject("plan");
       if (hasKey(plan, "queryPlan")) {
         final JSONObject queryPlan = plan.getJSONObject("queryPlan");
-        final String queryId = queryPlan.getString("queryId");
-        streamsProperties.put(
-            StreamsConfig.APPLICATION_ID_CONFIG,
-            QueryApplicationId.build(ksqlConfig, true, new QueryId(queryId)));
-
+        queryId = queryPlan.getString("queryId");
+        if (hasKey(queryPlan, "runtimeId")
+            && ((Optional<String>) queryPlan.get("runtimeId")).isPresent()) {
+          streamsProperties.put(
+              StreamsConfig.APPLICATION_ID_CONFIG,
+              ((Optional<String>) queryPlan.get("runtimeId")).get());
+          sharedRuntimeQuery = true;
+        } else {
+          streamsProperties.put(
+              StreamsConfig.APPLICATION_ID_CONFIG,
+              QueryApplicationId.build(ksqlConfig, true, new QueryId(queryId)));
+        }
         queryIdFound = true;
       }
     }
 
     // the command contains a query, clean up it's internal state store and also the internal topics
     if (queryIdFound) {
       final StreamsConfig streamsConfig = new StreamsConfig(streamsProperties);
-      final String applicationId =
-          streamsConfig.getString(StreamsConfig.APPLICATION_ID_CONFIG);
+      String topicPrefix = streamsConfig.getString(StreamsConfig.APPLICATION_ID_CONFIG);;
```

Comment:
nit: use ternary operator ( the xx ? yy : zz thing) to set this so we can make it `final`, or if/else with separate declaration, whatever